### ISSUE TITLE
Unable to login to icloud account

### ISSUE THREAD

**Issue by br0nes** [gcobb321/icloud3#380]
Starting two days ago, my integration stopped working and unable to login to my icloud account. I can confirm that I can login to my icloud account directly, so my username and password are correct. 

I get the following error message:
- Errors Starting iCloud3
- iCloud acct not logged into, invalid username/password

iCloud3 v3 - 3.0.5.9
Home Assistant
Core 2024.10.3
Supervisor 2024.10.2
Operating System 13.2
Frontend 20241002.3

**Comment by naseemr** [comment]
Same issue for me as well.

**Comment by GGSSDD** [comment]
Same problem here

**Comment by meiersmarkus** [comment]
Same here with HA v2024.7.0

**Comment by silke-99** [comment]
the native HA integration has similar issues, probably same story:  https://github.com/home-assistant/core/issues/128830

**Comment by fenen001** [comment]
Same here

**Comment by gcobb321** [comment]
This is a known issue that duplicates Issue #341 [here](https://github.com/gcobb321/icloud3_v3/issues/341)

**Comment by cedricdelecole** [comment]
here is a screenshot of the issue just in case 👍 
![image](https://github.com/user-attachments/assets/84cd9ba2-ed2d-41ed-b8c2-3d8222dbfe19)

**Comment by ch155260** [comment]
I had same problem, but I'm only login an account.

**Comment by gcobb321** [comment]
THANKS FOR LETTING ME KNOW THAT YOU ARE ALSO SEEING THE PROBLEM BUT THAT IS NOT NECESSARY. Everyone is having this problem, including the HA iCloud integration, people that use the Python Pyicloud interface and many other apps all around the world.

Apple changed to a new protocol to verify an account password where a secure hash key is passed from the client (iCloud3) to the Apple servers instead of the password. It is called Secure Remote Password. It is a complicated handshake involving a lot of back-and-forth between the client and Apple. It is working on one Ruby program we know of and we are working to port that to Python. Until it is complete, tested and implemented, you are going to have to use the Mobile App to locate your devices.

I’ll let you know when it is ready. Stay tuned.

**Comment by fakezeta** [comment]
Just to share update for the people coming here and avoid questioning. 
A PR in the upstream pyicloud has been opened
https://github.com/picklepete/pyicloud/pull/460

**Comment by c0d3m0nky** [comment]
This PR seems to be the one getting community attention on the matter

https://github.com/picklepete/pyicloud/pull/459

**Comment by gcobb321** [comment]
Thanks for the heads up on this issue. When scrolling through the complete post, you will see me on a number of posts. I have been involved in the solution to this problem since the beginning. The v3.1 update was a result. It is now up to v3.1.3 with a few bug fixes.

**Comment by c0d3m0nky** [comment]
Yeah, sorry, didn't mean to bug you on it, I was posting it for anyone else like me coming here thrice weekly hoping to see some news lol

**Comment by gcobb321** [comment]
This issue should actually be closed since the original issue has been fixed.

**Comment by gcobb321** [comment]
Closing due to no activity and the release of v3.1.5
